WordPress/Classes/ViewRelated/Domains/Domain registration/RegisterDomainSuggestions/RegisterDomainCoordinator.swift

```diff
@@ -141,20 +134,39 @@ class RegisterDomainCoordinator {
 
     // MARK: Helpers
 
+    private func purchaseDomain(for site: Blog?, on viewController: UIViewController, completion: @escaping (Result<Void, Swift.Error>) -> Void) {
+        self.createCart { [weak self] result in
+            guard let self else {
+                return
+            }
+            switch result {
+            case .success(let domain):
+                if let site, site.canRegisterDomainWithPaidPlan {
+                    self.domainAddedToCartAndLinkedToSiteCallback?(viewController, domain.domainName, site)
+                } else {
+                    self.presentCheckoutWebview(on: viewController, title: TextContent.checkoutTitle)
+                }
+                completion(.success(()))
+            case .failure(let error):
+                viewController.displayActionableNotice(title: TextContent.errorTitle, actionTitle: TextContent.errorDismiss)
```

Comment:
Shouldn't the `completion` block be called on an action of `displayActionableNotice`?

Reply:
Resolved in a DM.